Localization Scripts for Hub UI

adityamulik opened the pull request:
Hello,

The pull request is for the purpose of automating the localization of Ansible-Hub-UI. 
We're actively developing an Ansible collection for this process called as [Ansible Memsource Collection](https://github.com/ansible/ansible-collection-memsource)

The process comprises of two tasks:
- Pull Strings from the project (Using the project framework's strings extraction command). These strings (Usually retrieved are in English) are then uploaded to a third party software called Memsource
- Push Strings to the projects, once the translations are completed and the strings are ready to be pushed back to the repository, the push/ post_translation role of this collection is executed

For this process, we require two shell scripts to be placed in the repository. 
- pre_translation.sh - which performs the extraction of strings from a project. In this process, a folder called "translations" will be provided, the script ensures to move the strings to the translations folder
   - Empty translations folder
- post_translation.sh - which performs the process of moving the strings to appropriate directory in the project, a PR is later autogenerated by the underlying Ansible role being utilized here for merging the translated strings. Again a translations folder will be provided with all the translated strings, the script would ensure to utilize the folder and move the files according to the project needs.
   - Translation folder with all languages generated files
   e.g. 
          /translations/fr/en.po 
          /translations/jp/en.po
          /translations/ko/en.po

himdel commented:
@adityamulik do these changes belong in this repo?

Most of the script seems concerned with:
* switching to hardcoded `_clones/ansible-hub-ui`, without showing how to create it
* copying files back and forth between `locale/` and `translations/`, without showing how `translations/` gets created or what happens in between
* renaming one specific locale form a long form to a short one, but we never use the long version here

Seems like all these are very interdependent part of some larger toolchain, so maybe they should live there?
(Or, if you really want to add them here, we should at least add a comment with a link to more info, whether it's the repo using it, interface docs, or a jira ticket describing the purpose.)

Can you also write a PR description please?

himdel commented on tools/scripts/l18n/post_translation.sh:
```diff
@@ -0,0 +1,16 @@
+cd _clones/ansible-hub-ui/
+
+# Rename the zh_cn folder 
+mv translations/zh_cn translations/zh
```
if `translations/zh_cn/` is a folder, then `locale/zh/` is also a folder at the end

that's the wrong format, we expect a `locale/zh.po` file

himdel commented on tools/scripts/l18n/pre_translation.sh:
```diff
@@ -0,0 +1,12 @@
+# Move files to translations folder
+mv locale/en.po translations/
+
+sleep 5
```
(nothing started here is asynchronous, the sleep should not be necessary)

adityamulik replied:
sleep command has been removed
4724a76bc00d9e09d89d0b45097e578a61fa044e

himdel commented on tools/scripts/l18n/pre_translation.sh:
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash/
```
```suggestion
#!/bin/bash
```

adityamulik replied:
trailing slash has been removed.
4724a76bc00d9e09d89d0b45097e578a61fa044e

adityamulik commented:
Hi @himdel,

I've updated the description of the PR and the purpose of these two translation shell scripts.

1. For the _clones/ansible-hub-ui name, I have opened an issue in the ansible-memsource-collection project to fix the static path being used and change to dynamic. Similarly, I've removed the static path from the next commit as the path was redundant. 
Ref: https://github.com/ansible/ansible-collection-memsource/issues/25
Ref: 4724a76bc00d9e09d89d0b45097e578a61fa044e

2. I've updated the script to have the files as per the project's current structure. 
We would be receiving file from Memsource in below structure.

e.g.
/translations/fr/en.po
/translations/jp/en.po
/translations/ko/en.po

The script would modify the files and sync it to the required location
/translations/fr/en.po -> /locale/fr.po

3. We also receive the lang zh as zh_cn from Memsource, so the script ensures to first move the folder from zh_cn -> zh
Then move the file zh/en.po -> zh.po

We have two underlying roles for localization
1. https://github.com/ansible/ansible-collection-memsource/blob/main/roles/pre_translation/README.md
2. https://github.com/ansible/ansible-collection-memsource/blob/main/roles/post_translation/README.md

rooftopcellist commented:
@himdel One benefit to having these scripts live in the repo with the code is that if the structure changes over time, the script can be modified on that branch / versioned with the code it acts on.  

But I've added the ability to specify a script local to the memsource roles which gets copied into the clone of the app (in this case ansible-hub-ui), so we can close this PR and move forward with that.  
* https://github.com/ansible/ansible-collection-memsource/pull/45

adityamulik commented:
Closing the PR as applicable local script changes has been done as per https://github.com/ansible/ansible-collection-memsource/pull/45
